### PR TITLE
ARROW-10310: [C++][Gandiva] Add single argument round() in Gandiva

### PR DESCRIPTION
**Pull request by sgnkc**

### CODE REVIEW

File: cpp/src/gandiva/precompiled/extended_math_ops.cc [hunk 1]
```diff
@@ -122,8 +138,8 @@ POWER(float64, float64, float64)
         scale_multiplier);                                                  \
   }
 
-ROUND_DECIMAL(float32)
-ROUND_DECIMAL(float64)
+ROUND_DECIMAL_TWO_ARG(float32)
```
Comment (pprudhvi) [inline cpp/src/gandiva/precompiled/extended_math_ops.cc:141]: can we rename this to say something like 'round_decimal_to_scale'?

File: cpp/src/gandiva/precompiled/extended_math_ops.cc [hunk 2]
```diff
@@ -112,8 +112,24 @@ LOG_WITH_BASE(float64, float64, float64)
...
+gdv_int64 round_int64(gdv_int64 num) { return num; }
+
+// one-argument round
+#define ROUND_DECIMAL_ONE_ARG(TYPE)                                        \
+  FORCE_INLINE                                                             \
+  gdv_##TYPE round_##TYPE(gdv_##TYPE num) {                                \
+    return static_cast<gdv_##TYPE>(trunc(num + ((num > 0) ? 0.5 : -0.5))); \
+  }
+
+ROUND_DECIMAL_ONE_ARG(float32)
```
Comment (pprudhvi) [inline cpp/src/gandiva/precompiled/extended_math_ops.cc:128]: let this be just 'round_decimal'

File: cpp/src/gandiva/precompiled/extended_math_ops.cc [hunk 3]
```diff
@@ -112,8 +112,24 @@ LOG_WITH_BASE(float64, float64, float64)
...
+ROUND_DECIMAL_ONE_ARG(float64)
+
+// two-argument round
```
Comment (pprudhvi) [inline cpp/src/gandiva/precompiled/extended_math_ops.cc:131]: could you change the comment to say that this rounds to the given scale?

**Comment by pprudhvi** [comment]
LGTM, added comments to rename macros.
@praveenbingo , could you please take a look and merge?

File: cpp/src/gandiva/precompiled/extended_math_ops.cc (hunk 1 above)
Comment (sagnikc-dremio) [reply cpp/src/gandiva/precompiled/extended_math_ops.cc]: Addressed the comments.

**Review by praveenbingo — APPROVED** [review]
lgtm

**Pull request closed without merging**